### PR TITLE
RSC creating account for RSC generates invitation to register

### PR DESCRIPTION
**Pull request by RhonGabriel**
#PT 
https://www.pivotaltracker.com/story/show/161645003

### CODE REVIEW

File: app/views/devise/invitations/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  <% if f.object.class.require_password_on_accepting %>
+  <p><%= f.label :password %><br />
+  <%= f.password_field :password %></p>
+
+  <p><%= f.label :password_confirmation %><br />
+  <%= f.password_field :password_confirmation %></p>
+  <% end %>
+
+  <p><%= f.submit t("devise.invitations.edit.submit_button") %></p>
+<% end %>
```
Comment (Prozak8) [inline app/views/devise/invitations/edit.html.erb]: convert to haml

File: config/environments/development.rb [hunk 2]
```diff
@@ -35,6 +35,14 @@
   # Don't care if the mailer can't send.
   config.action_mailer.raise_delivery_errors = false
 
+  config.action_mailer.delivery_method = :letter_opener  
+  config.action_mailer.default_url_options = { host: 'localhost' }
+
+  # config.action_mailer.delivery_method = :smtp
+
+  config.action_mailer.perform_deliveries = true
+  config.action_mailer.smtp_settings = { :address => "localhost", :port => 1025 }
```
Comment (Prozak8) [inline config/environments/development.rb]: comment and if no difference remove. New hash syntax

File: config/environments/production.rb [hunk 3]
```diff
@@ -84,7 +84,14 @@
...
   # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new 'app-name')
-
+  config.action_mailer.smtp_settings = {
+    :address              => "smtp.gmail.com",
+    :port                 => 587,
+    :user_name            => ENV['gmail_username'],
+    :password             => ENV['gmail_password'],
+    :authentication       => "plain",
+    :enable_starttls_auto => true
+}
```
Comment (Prozak8) [inline config/environments/production.rb]: remove

File: config/environments/test.rb [hunk 4]
```diff
@@ -30,13 +30,17 @@
...
 
   config.action_mailer.perform_caching = false
 
   # Tell Action Mailer not to deliver emails to the real world.
   # The :test delivery method accumulates sent emails in the
   # ActionMailer::Base.deliveries array.
-  config.action_mailer.delivery_method = :test
+  # config.action_mailer.delivery_method = :cache
+  config.action_mailer.perform_deliveries = false
+  config.action_mailer.delivery_method = :letter_opener
```
Comment (Prozak8) [inline config/environments/test.rb:43]: remove this one line

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -1,6 +1,8 @@
 require 'coveralls'
 Coveralls.wear_merged!('rails')
 require 'spec_helper'
+require "email_spec"
+require "email_spec/rspec"
```
Comment (Prozak8) [inline spec/rails_helper.rb]: not testing it in rspec only in cucumber

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -1,8 +1,17 @@
+require "action_mailer"
+require "email_spec"
```
Comment (Prozak8) [inline spec/spec_helper.rb]: no need

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -1,8 +1,17 @@
...
 
   config.mock_with :rspec do |mocks|
   end
+
+  config.include EmailSpec::Helpers
+  config.include EmailSpec::Matchers
+
+  config.before(:each) do
+    reset_mailer
+  end
```
Comment (Prozak8) [inline spec/spec_helper.rb]: remove this

File: Gemfile [hunk 8]
```diff
@@ -29,6 +30,9 @@ group :development, :test do
   gem 'database_cleaner'
   gem 'chromedriver-helper'
   gem 'selenium-webdriver'
+  gem 'email_spec'
+  gem 'letter_opener'
+  gem 'action_mailer_cache_delivery'
```
Comment (Prozak8) [inline Gemfile]: do we need this?

File: app/controllers/registrations_controller.rb [hunk 9]
```diff
@@ -21,5 +21,20 @@ def create
...
+      if self.resource && self.resource.respond_to?(:invited_to_sign_up?) && self.resource.invited_to_sign_up?
+        self.resource.attributes = hash
+        self.resource.send_confirmation_instructions if self.resource.confirmation_required_for_invited?
+        self.resource.accept_invitation
+      else
+        self.resource = nil
+      end
+    end
+    self.resource ||= super
+  end
```
Comment (Prozak8) [inline app/controllers/registrations_controller.rb]: what is this doing?

File: spec/spec_helper.rb (hunk 6 above)
Comment (Noelryn) [reply spec/spec_helper.rb]: Removed these!

File: spec/spec_helper.rb (hunk 7 above)
Comment (Noelryn) [reply spec/spec_helper.rb]: This is removed !!!

File: Gemfile (hunk 8 above)
Comment (Noelryn) [reply Gemfile]: Removed action_mailer_cache_delivery

**Pull request merged**